lib/core/test/unit/Cardano/Wallet/TransactionSpecShared.hs

```diff
@@ -0,0 +1,65 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+module Cardano.Wallet.TransactionSpecShared
+    ( propMaxNumberOfInputsEstimation
+    ) where
+
+import Prelude
+
+import Cardano.Wallet.Transaction
+    ( TransactionLayer (..) )
+import Data.Quantity
+    ( Quantity (..) )
+import Data.Word
+    ( Word16, Word8 )
+import Test.QuickCheck
+    ( Arbitrary (..)
+    , Property
+    , Small (..)
+    , choose
+    , counterexample
+    , elements
+    , (.&&.)
+    )
+
+{-------------------------------------------------------------------------------
+                             Max inputs estimation
+-------------------------------------------------------------------------------}
+
+propMaxNumberOfInputsEstimation
+    :: TransactionLayer t
+    -> Quantity "byte" Word16
+    -> Quantity "byte" Word16
+    -> Word8
+    -> Word8
+    -> Property
+propMaxNumberOfInputsEstimation tl qa@(Quantity ma) qb@(Quantity mb) oa ob =
+    counterexample debug
+    (isIncreasingFunction .&&. moreOutputsLessInputs .&&. estIsSmallerThanSize)
```

Comment:
In general, I'd be more in favor of having three properties here instead of a conjunction of assertions. The error reporting from QuickCheck isn't ideal with `.&&.` (or `conjoin`), whereas, with separate properties, we immediately know which property is failing.